docs/pages/auth-concept/authentication.md

```diff
@@ -0,0 +1,181 @@
+---
+title: 'Authentication functions'
+weight: 3
+---
+
+# Getting started with Authentication functions
+
+Authentication in Glee can be done using authentication functions. Authentication functions are files that export either one or both of the `clientAuth` and `serverAuth` Node.js functions:
+
+```js
+/* websocket.js */
+
+export async function serverAuth({ authProps, done }) {
+  //server auth logic
+}
+
+export async function clientAuth({ parsedAsyncAPI, serverName }) {
+  //client auth logic
+}
+```
+
+Glee looks for authentication files in the `auth` directory by default but it can be configured using [glee config file](../config-file.md).
+The name of the authentication file should be the name of the targeted server that the authentication logic should work for.
+
+## Supported Authentication Values in asyncapi.yaml file
+
+AsyncAPI currently supports a variety of authentication formats as specified in the [documentation](https://www.asyncapi.com/docs/reference/specification/v2.6.0#securitySchemeObject), however Glee supports the following authentication schemas.
+
+- userPassword
+- Bearer token
+- HttpPApiKey
+- ApiKey
+
+A sample `asyncapi.yaml` for a server with security requirements and a `userPassword` security schemes is shown below:
+
+```yaml
+##server asyncAPI schema
+asyncapi: 2.6.0
+info:
+  title: AsyncAPI IMDB server
+  version: 1.0.0
+  description: This app is a dummy server that would stream the trending/upcoming anime.
+servers:
+  trendingAnimeServer:
+    url: 'http://localhost:8081'
+    protocol: http
+    security:
+      - userPass: []
+
+  ...
+
+components:
+  securitySchemes:
+    userPass:
+      type: userPassword
+
+```
+
+A sample `asyncapi.yaml` for a client that implements some of the requirements of the server above:
+
+```yaml
+##client asyncAPI schema
+servers:
+  trendingAnime:
+    url: http://localhost:8081
+    protocol: http
+    security:
+      - userPass: []
+  testwebhook:
+    url: ws://localhost:9000
+    protocol: ws
+x-remoteServers:
+  - trendingAnime
+
+  ...
+
+components:
+  securitySchemes:
+    userPass:
+      type: userPassword
+
+```
+
+**The Client asyncapi.yaml file does not need to implement all the security requirements in the server, it only needs to implement the ones that it uses.**
+
+
+Glee can act as both a server and a client at the same time. Hence the need for `serverAuth` and `clientAuth`. Glee acts as a client when the `x-remoteServers` property is present in the `asyncapi.yaml` file.
+
+When Glee acts as a client, it can connect to a Glee server, and when Glee acts as a server it accepts connection from other Glee clients. Hence a Glee application can both accept connections from clients while also sending requests to other Glee applications (servers) ath the same time.
```

Comment:
```suggestion
When Glee acts as a client, it can connect to a Glee server, and when Glee acts as a server it accepts connections from other Glee clients. Hence a Glee application can both accept connections from clients while also sending requests to other Glee applications (servers) at the same time.
```